filebeat/harvester/harvester.go

```diff
@@ -17,12 +17,15 @@ import (
 	"fmt"
 	"regexp"
 
+	"github.com/satori/go.uuid"
+
 	"github.com/elastic/beats/filebeat/config"
 	"github.com/elastic/beats/filebeat/harvester/encoding"
 	"github.com/elastic/beats/filebeat/input"
 )
 
 type Harvester struct {
+	Id                 uuid.UUID
```

Comment:
The goal is to be able to track harvesters and also properly shut them down. So in case an error appears it is possible to better tell which harvester had problems. This is related to the PR here: https://github.com/elastic/beats/pull/964/files#diff-72613c7cf09182c67339cf99ed741eccR25